src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.CSharp.Tests.cs

```diff
@@ -0,0 +1,488 @@
+        return 4;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        if (a.StartsWith(b, StringComparison.InvariantCultureIgnoreCase))
+        {
+            return 5;
+        }
+        return 4;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 13, 8, 47, StringComparisonRule, StartsWithName);
+        }
+
+        [Fact]
+        public Task Diagnostic_CompareTo_ToLower_Assign()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = a.ToLower().CompareTo(b);
+        return r;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = StringComparer.CurrentCultureIgnoreCase.Compare(a, b);
+        return r;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 17, 8, 41, StringComparerRule, CompareToName);
+        }
+
+        [Fact]
+        public Task Diagnostic_CompareTo_ToLowerInvariant_Assign()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = a.ToLowerInvariant().CompareTo(b);
+        return r;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = StringComparer.InvariantCultureIgnoreCase.Compare(a, b);
+        return r;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 17, 8, 50, StringComparerRule, CompareToName);
+        }
+
+        [Fact]
+        public Task Diagnostic_CompareTo_ToUpper_Assign()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = a.ToUpper().CompareTo(b);
+        return r;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = StringComparer.CurrentCultureIgnoreCase.Compare(a, b);
+        return r;
+    }
+}";
+            return VerifyCSharpAsync(originalCode, fixedCode, 8, 17, 8, 41, StringComparerRule, CompareToName);
+        }
+
+        [Fact]
+        public Task Diagnostic_CompareTo_ToUpperInvariant_Assign()
+        {
+            string originalCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = a.ToUpperInvariant().CompareTo(b);
+        return r;
+    }
+}";
+            string fixedCode = @"using System;
+class C
+{
+    int M()
+    {
+        string a = ""aBc"";
+        string b = ""bc"";
+        int r = StringComparer.InvariantCultureIgnoreCase.Compare(a, b);
```

Comment:
Consider testing named arguments, also with the order swapped (I don't expect this to be in real code though)

Reply:
Yep. It's in my TODO list in the description.

Reply:
Addressed! Phew that was... interesting.